File: CS50_Week_6/CS50_problemset6/sentimental-readability/readability.py
# Outputs grade level required to read some text entered by the user.

# Get text from the user
text = input("Text: ")

# Tracks the number of letters, words, and sentences in text respectively
letters = 0
words = 0
sentences = 0

# Iterate through each character in text and track how many letters, words, and sentences there are in it.
for i in text:
    # If a letter is detected letters++
    if i.isalpha():
        letters += 1
    # If a space is detected words ++
    elif i == ' ':
        words += 1
    # If ending punctuation is detected sentences++
    elif i == '.' or i == '!' or i == '?':
        sentences += 1

# Add last word in text to words
words += 1

# Use variables to calculate reading level
index = 0.0588 * (letters / words * 100) - 0.296 * (sentences / words * 100) - 15.8
index = round(index)

# Output level to user
if index >= 1 and index <= 16:
    print(f"Grade {index}")
elif index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")

File: CS50_Week_2/CS50_problemset2/readability.c
// Calculates the approximate grade level needed to comprehend some text.

#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // Declare and initialize variables.
    float letters = 0;
    float words = 0;
    float sentences = 0;

    // Get text from the user.
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text.
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        else if (text[i] == ' ')
        {
            words++;
        }
        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }
    // Detecting words by number of spaces, this prevents the code from not counting the last word in text.
    words++;

    // Calculate reading level.
    float index = 0.0588 * (letters / words * 100) - 0.296 * (sentences / words * 100) - 15.8;
    index = round(index);

    // Print reading level to the user.
    if (index >= 1 && index <= 16)
    {
        printf("Grade %i\n", (int) index);
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }
}
